sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/metric/viewconfig/ViewConfig.java

```diff
@@ -192,6 +213,41 @@ static View toView(ViewSpecification viewSpec) {
     return builder.build();
   }
 
+  static View toView(ViewSpecification viewSpec,@Nullable ConfigProperties configProperties) {
+    ViewBuilder builder = View.builder();
+    // add additional attributes
+    if (configProperties != null) {
+      AttributesBuilder attributesBuilder = Attributes.builder();
+      for (ConfigurableMetricAttributesProvider provider :  ServiceLoader.load(
+          ConfigurableMetricAttributesProvider.class)){
+        attributesBuilder.putAll(provider.addCustomAttributes(configProperties));
+      }
+      SdkMeterProviderUtil.appendAdditionalAttributes(builder, AttributesProcessor.append(attributesBuilder.build()));
```

Comment:
Since this is an extension to the YAML file configuration for views, I think it would make more sense to implement this as an additional section of key value pairs within the YAML, rather than an SPI that surprisingly interacts with all registered views. 

With that said, I'm reluctant to add configurable surface area for adding these attributes since using views to add additional attributes is experimental and not an official part of the spec. That means that when we replace this file based YAML config with the [full file configuration](https://github.com/open-telemetry/opentelemetry-java/pull/5399) solution (currently being worked on), we'd have to drop the ability to append attributes via views.

Reply:
> Since this is an extension to the YAML file configuration for views, I think it would make more sense to implement this as an additional section of key value pairs within the YAML, rather than an SPI that surprisingly interacts with all registered views.
> 
> With that said, I'm reluctant to add configurable surface area for adding these attributes since using views to add additional attributes is experimental and not an official part of the spec. That means that when we replace this file based YAML config with the [full file configuration](https://github.com/open-telemetry/opentelemetry-java/pull/5399) solution (currently being worked on), we'd have to drop the ability to append attributes via views.

Agree, My earliest consideration was the implementation you mentioned. Perhaps a specific metric  needs to extend attributes,that's very suitable,Considering the common attributes of metrics, such as clusters, planes, etc., If each metric needs to be configured separately, it would be a lot of work,So I adopted the SPI. Do you have any good suggestions for the common attributes of all metrics, There maybe be better way. @jack-berg

Reply:
Mentioned this in slack, but the most correct solution is probably to create a delegating MetricExporter which enriches MetricData with the additional attributes before calling a subsequent MetricExporter. While probably the most correct solution, this is awkward because it requires you to provide your own implementations of the MetricData interfaces. There's an issue tracking this [here](https://github.com/open-telemetry/opentelemetry-java/issues/4389).